ext/pgsql/pgsql.c

```diff
@@ -6248,3 +6246,29 @@ PHP_FUNCTION(pg_socket_poll)
 
 	RETURN_LONG((zend_long)PQsocketPoll(socket, (int)read, (int)write, (int)ts));
 }
+
+#if defined(HAVE_PG_SET_CHUNKED_ROWS_MODE)
+PHP_FUNCTION(pg_set_chunked_rows_mode)
+{
+	zval *pgsql_link;
+	pgsql_link_handle *link;
+	zend_long size;
+
+	ZEND_PARSE_PARAMETERS_START(2, 2)
+		Z_PARAM_OBJECT_OF_CLASS(pgsql_link, pgsql_link_ce)
+		Z_PARAM_LONG(size)
+	ZEND_PARSE_PARAMETERS_END();
+
+	if (size < 1 || size > INT_MAX) {
+		zend_argument_value_error(2, "must be between 1 and %d", INT_MAX);
+		RETURN_THROWS();
+	}
+
+	link = Z_PGSQL_LINK_P(pgsql_link);
+	CHECK_PGSQL_LINK(link);
+
+	/** can still fail if it is not allowed e.g. already fetched results **/
+
+	RETURN_LONG((zend_long)PQsetChunkedRowsMode(link->conn, (int)size));
```

Comment:
What value is returned on failure? `0` ?

Reply:
yes. it might be more appropriate to return a bool then.